### ISSUE TITLE
 [a11y]:  Combobutton [with danger] - screen reader sequence announcement bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15261]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.40.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.40.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
In the Combo Button , there are 3  Items , here  sequence number for  the "Danger action Item sequence announcement 1  of 1  is not correct , as well as  creating  confusion.
JAWS speech History  shared below :
Second action 
1 of 3
Third action 
2 of 3
Danger action 
1 of 1

Please confirm the expected behavior .
Make sure the sequence number of the  mentioned  Combo option , announced  correctly , that is  "Danger action  3  of 3  "
NOTE: We have understood from the call There is divider between Danger option and other options hence recommended to provide Group Label for the benefit of SR users.

### Recommended Fix
When Options in the drop down of the same logical group are not categorized together and given a group label, the user may not be able to understand the context and relevance of those fields. This defect mainly impacts users who are blind or have cognitive disabilities. 

Group Label's needs to be provided for the benefit of SR users

### WCAG 2.1 Violation
1.3.1 Info and Relationships

**Comment by 2nikhiltom** [comment]
Re-tested and confirmed on v11.84.0 as well
